pytorch_lightning/trainer/training_loop.py

```diff
@@ -685,71 +693,86 @@ def run_training_batch(self, batch, batch_idx, dataloader_idx):
+            training_step_output_for_epoch_end=batch_outputs,
+        )
+        return output
 
-                # ------------------------------
-                # BACKWARD PASS
-                # ------------------------------
-                # gradient update with accumulated gradients
-                else:
-                    if self.trainer.lightning_module.automatic_optimization:
+    def run_optimization(self, batch_idx, split_idx, split_batch, opt_idx=None, optimizer=None):
+        # toggle model params + set info to logger_connector
+        self.run_train_split_start(split_idx, split_batch, opt_idx, optimizer)
 
-                        def train_step_and_backward_closure():
-                            nonlocal result
-                            result = self.training_step_and_backward(
-                                split_batch, batch_idx, opt_idx, optimizer, self.trainer.hiddens
-                            )
-                            return None if result is None else result.loss
+        result = AttributeDict()
+        closure = self.make_closure(split_batch, batch_idx, opt_idx, optimizer, self.trainer.hiddens, result)
 
-                        # optimizer step
-                        self.optimizer_step(optimizer, opt_idx, batch_idx, train_step_and_backward_closure)
+        if self.should_accumulate():
+            # For gradient accumulation
 
-                    else:
-                        result = self.training_step(split_batch, batch_idx, opt_idx, self.trainer.hiddens)
+            # -------------------
+            # calculate loss (train step + train step end)
+            # -------------------
 
-                    if not result:
-                        # user decided to skip optimization
-                        # make sure to zero grad.
-                        continue
+            # automatic_optimization=True: perform dpp sync only when performing optimizer_step
+            # automatic_optimization=False: don't block synchronization here
+            with self.block_ddp_sync_behaviour():
+                closure()
 
-                    # todo: Properly aggregate grad_norm accros opt_idx and split_idx
-                    grad_norm_dict = result.get("grad_norm_dict", {})
+        # ------------------------------
+        # BACKWARD PASS
+        # ------------------------------
+        # gradient update with accumulated gradients
+        else:
+            if self.trainer.lightning_module.automatic_optimization:
+                self.optimizer_step(optimizer, opt_idx, batch_idx, closure)
+            else:
+                result = self.training_step(split_batch, batch_idx, opt_idx, self.trainer.hiddens)
 
-                    # update running loss + reset accumulated loss
-                    self.update_running_loss(result.loss)
+            if not result:
+                # user decided to skip optimization
+                # make sure to zero grad.
```

Comment:
This comment is outdated. It comes from https://github.com/PyTorchLightning/pytorch-lightning/pull/4485/files#diff-6b21474ed45079f01dfa45ee3b9d40d23efe693c9c005ce897e25384ef425349R701-R702

```suggestion
```